x-pack/elastic-agent/pkg/agent/control/server/server.go

```diff
@@ -210,70 +213,232 @@ func (s *Server) ProcMeta(ctx context.Context, _ *proto.Empty) (*proto.ProcMetaR
 		specs := sp.Specs()
 
 		for n, spec := range specs {
-			procMeta := &proto.ProcMeta{
-				Name:     n,
-				RouteKey: rk,
-			}
-
-			client := http.Client{
-				Timeout: time.Second * 5,
-			}
 			endpoint := monitoring.MonitoringEndpoint(spec, runtime.GOOS, rk)
-			if strings.HasPrefix(endpoint, "unix://") {
-				client.Transport = &http.Transport{
-					Proxy:       nil,
-					DialContext: socket.DialContext(strings.TrimPrefix(endpoint, "unix://")),
-				}
-				endpoint = "unix"
-			} else if strings.HasPrefix(endpoint, "npipe://") {
-				client.Transport = &http.Transport{
-					Proxy:       nil,
-					DialContext: socket.DialContext(strings.TrimPrefix(endpoint, "npipe:///")),
-				}
-				endpoint = "npipe"
-			}
+			client := newSocketRequester(time.Second*5, n, rk, endpoint)
+
+			procMeta := client.ProcMeta(ctx)
+			resp.Procs = append(resp.Procs, procMeta)
+		}
+	}
+	return resp, nil
+}
+
+// Pprof returns /debug/pprof data for the requested applicaiont-route_key or all running applications.
+func (s *Server) Pprof(ctx context.Context, req *proto.PprofRequest) (*proto.PprofResponse, error) {
+	if s.routeFn == nil {
+		return nil, errors.New("route function is nil")
+	}
+
+	dur, err := time.ParseDuration(req.TraceDuration)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse trace duration: %w", err)
+	}
+
+	resp := &proto.PprofResponse{
+		Results: []*proto.PprofResult{},
+	}
+
+	var wg sync.WaitGroup
+	ch := make(chan *proto.PprofResult, 1)
+
+	// retrieve elastic-agent pprof data if requested or application is unspecified.
+	if req.AppName == "" || req.AppName == "elastic-agent" {
+		endpoint := beats.AgentDebugEndpoint(runtime.GOOS)
+		c := newSocketRequester(dur*2, "elastic-agent", "", endpoint)
```

Comment:
One more place where we have `*2`. Curious to understand this better.

Reply:
I think that the duration passed to `newSocketRequester` is an artifact of my development (I was not doing requests with contexts at one point); i'll clean this up